tests/FileStorageTest.php

```diff
@@ -14,25 +14,264 @@
 {
 
-    public function testFileStorageCreate()
-    {
-        $this->markTestSkipped('just for development');
-        $storage = $this->client->createFileStorage([
-           'awsKey' => '',
-           'awsSecret' => '',
-           'region' => 'eu-central-1',
-           'filesBucket' => 'martin-storage-3-s3filesbucket-jodg2rehxtvu',
-        ]);
+    private const DISABLE_DEV_TESTS = true;
+
+    private const DEFAULT_S3_OPTIONS = [
+        'awsKey' => TEST_AWS_KEY,
+        'awsSecret' => TEST_AWS_SECRET,
+        'region' => TEST_AWS_REGION,
+        'filesBucket' => TEST_AWS_FILES_BUCKET,
+        'owner' => 'keboola',
+    ];
+
+    private const DEFAULT_ABS_OPTIONS = [
+        'accountName' => TEST_ABS_ACCOUNT_NAME,
+        'accountKey' => TEST_ABS_ACCOUNT_KEY,
+        'containerName' => TEST_ABS_CONTAINER_NAME,
+        'owner' => 'keboola',
+    ];
+
+    private const ROTATE_S3_OPTIONS = [
+        'awsKey' => TEST_AWS_ROTATE_KEY,
+        'awsSecret' => TEST_AWS_ROTATE_SECRET,
+    ];
+
+    private const ROTATE_ABS_OPTIONS = [
+        'accountKey' => TEST_ABS_ROTATE_ACCOUNT_KEY,
+    ];
+
+    public function testFileStorageS3Create()
+    {
+        if (self::DISABLE_DEV_TESTS) {
+            $this->markTestSkipped('just for development');
+        }
+
+        $storage = $this->client->createS3FileStorage(self::DEFAULT_S3_OPTIONS);
 
-        $this->assertArrayHasKey('awsKey', $storage);
         $this->assertArrayNotHasKey('awsSecret', $storage);
-        $this->assertArrayHasKey('region', $storage);
-        $this->assertArrayHasKey('filesBucket', $storage);
+        $this->assertSame($storage['awsKey'], TEST_AWS_KEY);
+        $this->assertSame($storage['region'], TEST_AWS_REGION);
+        $this->assertSame($storage['filesBucket'], TEST_AWS_FILES_BUCKET);
+        $this->assertSame($storage['provider'], 'aws');
+        $this->assertSame($storage['isDefault'], false);
+    }
+
+    public function testFileStorageAbsCreate()
+    {
+        if (self::DISABLE_DEV_TESTS) {
+            $this->markTestSkipped('just for development');
+        }
+
+        $storage = $this->client->createAbsFileStorage(self::DEFAULT_ABS_OPTIONS);
+
+        $this->assertArrayNotHasKey('accountKey', $storage);
+        $this->assertSame($storage['accountName'], TEST_ABS_ACCOUNT_NAME);
+        $this->assertSame($storage['containerName'], TEST_ABS_CONTAINER_NAME);
+        $this->assertSame($storage['provider'], 'azure');
+        $this->assertSame($storage['isDefault'], false);
+    }
+
+    public function testRotateS3Key()
+    {
+        if (self::DISABLE_DEV_TESTS) {
+            $this->markTestSkipped('just for development');
+        }
+
+        $storage = $this->client->createS3FileStorage(self::DEFAULT_S3_OPTIONS);
+
+        $rotatedStorage = $this->client->rotateS3FileStorageCredentials($storage['id'], self::ROTATE_S3_OPTIONS);
+
+        $this->assertSame($rotatedStorage['awsKey'], TEST_AWS_ROTATE_KEY);
```

Comment:
asserty by měly být i na zbytek IMHO - že se něco změní a něco ne....

Reply:
doplnil jsem to tam.